Remotion/Web/Web.Development.WebTesting.IntegrationTests.RequireUI/MouseTest.cs

```diff
@@ -0,0 +1,165 @@
+    private const string c_focusDivID = "focusDiv";
+    private const string c_hoverDivID = "hoverDiv";
+
+    private const string c_getBackgroundColorJs =
+        @"return window.getComputedStyle (arguments[0])['background-color'].match (/rgb\(\s*([0-9]+),\s*([0-9]+),\s*([0-9]+)\)/).slice (1,4);";
+
+    [Test]
+    public void Test_CorrectDivColors ()
+    {
+      var home = Start();
+
+      var clickElement = home.Scope.FindId(c_clickDivID);
+      Assert.That(GetColor(clickElement).ToArgb(), Is.EqualTo(Color.Black.ToArgb()));
+
+      var focusElement = home.Scope.FindId(c_clickDivID);
+      Assert.That(GetColor(focusElement).ToArgb(), Is.EqualTo(Color.Black.ToArgb()));
+
+      var hoverElement = home.Scope.FindId(c_clickDivID);
+      Assert.That(GetColor(hoverElement).ToArgb(), Is.EqualTo(Color.Black.ToArgb()));
+    }
+
+    [Test]
+    public void Test_MouseLeftClick ()
+    {
+      var home = Start();
+
+      var clickElement = home.Scope.FindId(c_clickDivID);
+      Helper.BrowserConfiguration.MouseHelper.Hover(clickElement);
+      Helper.BrowserConfiguration.MouseHelper.LeftClick();
+      Thread.Sleep(250);
+
+      Assert.That(GetColor(clickElement).ToArgb(), Is.EqualTo(Color.Red.ToArgb()));
+    }
+
+    [Test]
+    public void Test_MouseDoubleLeftClick ()
+    {
+      var home = Start();
+
+      var clickElement = home.Scope.FindId(c_clickDivID);
+      Helper.BrowserConfiguration.MouseHelper.Hover(clickElement);
+      Helper.BrowserConfiguration.MouseHelper.DoubleLeftClick();
+      Thread.Sleep(250);
+
+      Assert.That(GetColor(clickElement).ToArgb(), Is.EqualTo(Color.Green.ToArgb()));
+    }
+
+    [Test]
+    public void Test_MouseRightClick ()
+    {
+      var home = Start();
+
+      var clickElement = home.Scope.FindId(c_clickDivID);
+      Helper.BrowserConfiguration.MouseHelper.Hover(clickElement);
+      Helper.BrowserConfiguration.MouseHelper.RightClick();
+      Thread.Sleep(250);
+
+      Assert.That(GetColor(clickElement).ToArgb(), Is.EqualTo(Color.Blue.ToArgb()));
+    }
+
+    [Test]
+    public void Test_Focus ()
+    {
+      var home = Start();
+
+      var focusElement = home.Scope.FindId(c_focusDivID);
+
+      Helper.BrowserConfiguration.MouseHelper.Hover(focusElement);
+      Helper.BrowserConfiguration.MouseHelper.LeftClick();
+
+      Thread.Sleep(250);
+
+      Assert.That(GetColor(focusElement).ToArgb(), Is.EqualTo(Color.Orange.ToArgb()));
+    }
+
+    [Test]
+    public void Test_Hover ()
+    {
+      var home = Start();
+
+      var hoverElement = home.Scope.FindId(c_hoverDivID);
+      Thread.Sleep(250);
+      Helper.BrowserConfiguration.MouseHelper.Hover(hoverElement);
+      Thread.Sleep(250);
+
+      Assert.That(GetColor(hoverElement).ToArgb(), Is.EqualTo(Color.Red.ToArgb()));
+    }
+
+    [Explicit("Testing the tooltip is flaky as long as there is no good implementation of .ShowTooltip")]
+    [Test]
+    public void Test_Tooltip ()
```

Comment:
This test requires saved screenshots so we need to move the folder over as well.